File: 2017-10-06-android/CleanArchitectureAndTDDDemo/core/src/test/java/com/example/interactors/createcat/CreateCatInteractorTests.java
package com.example.interactors.createcat;

import org.junit.Assert;
import org.junit.Test;

/**
 * Created by MyndDev on 06/10/2017.
 */
public class CreateCatInteractorTests {

    @Test
    public void shouldPassCorrectResponseToPresenter() {
        FakeCatCreatedPresenter presenter = new FakeCatCreatedPresenter();
        CreateCatInteractor interactor = new CreateCatInteractor(presenter);

        interactor.execute();

        Assert.assertEquals("Kang Kang 'The Great'", presenter.theReponseModelReceived().name);
    }
}

File: 2017-10-DevconDavao/CatsDemo_OutsideInTDD_DotNetCore/tests/Interactors.Tests/CreateCat/CreateCatInteractorTests.cs
using Interactors.CreateCat;
using System;
using Xunit;

namespace Interactors.Tests
{
    public class CreateCatInteractorTests
    {
        private FakeCreateCatPresenter presenter;
        private FakeCatRepository repository;

        private CreateCatRequest request;
        private CreateCatInteractor interactor;

        public CreateCatInteractorTests()
        {
            // arrange
            presenter = new FakeCreateCatPresenter();
            repository = new FakeCatRepository();
            request = new CreateCatRequest("Kang Kang", "The Great");
            interactor = new CreateCatInteractor(presenter, repository);

            // act
            interactor.Execute(request);
        }

        [Fact]
        public void ShouldSaveNewCatToTheDatabase()
        {
            // assert
            Assert.True(repository.TheSaveMethodWasCalled);
        }

        [Fact]
        public void ShouldCreateAResponseModel()
        {
            // assert
            Assert.NotNull(presenter.TheResponseModelPassedAsParameterToTheExecuteMethod.Id);
            Assert.Equal("Kang Kang", presenter.TheResponseModelPassedAsParameterToTheExecuteMethod.Name);
            Assert.Equal("The Great", presenter.TheResponseModelPassedAsParameterToTheExecuteMethod.Title);
        }

        [Fact]
        public void ShouldExecuteThePresenter()
        {
            // assert
            Assert.True(presenter.TheExecuteMethodWasCalled);
        }
    }
}
